src/main/scala/nl/knaw/dans/easy/archivebag/EasyArchiveBag.scala

```diff
@@ -70,17 +112,22 @@ object EasyArchiveBag {
   private def putFile(file: File)(implicit s: Parameters): CloseableHttpResponse = {
     val md5hex = computeMd5(file).get
     log.debug(s"Content-MD5 = $md5hex")
-    log.info(s"Sending bag to ${s.storageDepositService}, id = ${s.uuid}, with user = ${s.username}, password = ******")
-    val credsProv = new BasicCredentialsProvider
-    credsProv.setCredentials(new AuthScope(s.storageDepositService.getHost, s.storageDepositService.getPort), new UsernamePasswordCredentials(s.username, s.password))
-    val http = HttpClients.custom.setDefaultCredentialsProvider(credsProv).build()
+    log.info(s"Sending bag to ${s.storageDepositService}, id = ${s.uuid}, with user = ${s.username}, password = ****")
+    val http = createHttpClient(s.storageDepositService.getHost, s.storageDepositService.getPort, s.username, s.password)
     val put = new HttpPut(new URI(s.storageDepositService + s.uuid.toString))
     put.addHeader("Content-Disposition", "attachment; filename=bag.zip")
     put.addHeader("Content-MD5", md5hex)
     put.setEntity(new FileEntity(file, ContentType.create("application/zip")))
     http.execute(put)
   }
 
+  def createHttpClient(host: String, port: Int, user: String, password: String): CloseableHttpClient = {
+    val credsProv = new BasicCredentialsProvider
```

Comment:
@PaulBoon uses a library called ScalaJ-http for doing HTTP calls. See for example: https://github.com/DANS-KNAW/dccd-export/blob/master/src/main/scala/nl/knaw/dans/dccd/export/ExportUtils.scala. Have you considered this lib as well? I guess it looks nicer in our Scala code, since ScalaJ-http is written as a Scala API. I have not done a comparison in functionality, to be honest with you! But from what I have seen, it supports the stuff we need...